File: app/src/main/java/com/nrs/nsnik/notes/objects/CheckListObject.java
/*
 * Copyright (C) 2017 nsnikhil
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
 */

package com.nrs.nsnik.notes.objects;


import android.os.Parcel;
import android.os.Parcelable;

import java.io.Serializable;

public class CheckListObject implements Serializable, Parcelable {

    private static final Creator<CheckListObject> CREATOR = new Creator<CheckListObject>() {
        @Override
        public CheckListObject createFromParcel(Parcel in) {
            return new CheckListObject(in);
        }

        @Override
        public CheckListObject[] newArray(int size) {
            return new CheckListObject[size];
        }
    };
    private String mText;
    private boolean mDone;

    public CheckListObject(String text, boolean done) {
        mText = text;
        mDone = done;
    }

    private CheckListObject(Parcel in) {
        mText = in.readString();
        mDone = in.readByte() != 0;
    }


    public static Creator<CheckListObject> getCREATOR() {
        return CREATOR;
    }

    public String getmText() {
        return mText;
    }

    public void setmText(String mText) {
        this.mText = mText;
    }

    public boolean ismDone() {
        return mDone;
    }

    public void setmDone(boolean mDone) {
        this.mDone = mDone;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(mText);
        parcel.writeByte((byte) (mDone ? 1 : 0));
    }
}

File: app/src/main/java/com/nrs/nsnik/notes/model/CheckListObject.kt
/*
 * Copyright (C) 2017 nsnikhil
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
 */

package com.nrs.nsnik.notes.model


import com.twitter.serial.serializer.ObjectSerializer
import com.twitter.serial.serializer.SerializationContext
import com.twitter.serial.stream.SerializerInput
import com.twitter.serial.stream.SerializerOutput
import java.io.Serializable


class CheckListObject : Serializable {

    var text: String = ""
    var done: Boolean = false

    companion object {

        val SERIALIZER: ObjectSerializer<CheckListObject> = CheckListObjectSerializer()

        class CheckListObjectSerializer : ObjectSerializer<CheckListObject>() {

            override fun serializeObject(context: SerializationContext, output: SerializerOutput<out SerializerOutput<*>>, checkListObject: CheckListObject) {
                output.writeString(checkListObject.text)
                output.writeBoolean(checkListObject.done)
            }

            override fun deserializeObject(context: SerializationContext, input: SerializerInput, versionNumber: Int): CheckListObject? {
                val checkListObject = CheckListObject()
                checkListObject.text = input.readString()!!
                checkListObject.done = input.readBoolean()
                return checkListObject
            }
        }
    }
}
